src/AI/CoreLogic.pm

```diff
@@ -1059,16 +1057,25 @@ sub processTransferItems {
 				#need to low down the amount
 				$row->{amount} = $freeWeight / ( $item->weight()/10 );
 				warning TF("Amount of %s is more than you can carry, getting the maximum possible (%d)\n", $row->{item}, $row->{amount});
-			}		
+			}
 		}
 
 		if ( $row->{source} eq 'inventory' && $item->{equipped} ) {
 			error TF( "Inventory item '%s' is equipped.\n", $item->{name} );
 			redo;
 		}
-
+		
+		my $target_item = $target->getByName( $row->{item}->name );
+		
 		# Transfer the item!
-		$messageSender->$method( $item->{ID}, min( $item->{amount}, $row->{amount} || $item->{amount} ) );
+		if ((30000 - $target_item->{amount} ) <= 0) {
```

Comment:
Is 30k ALWAYS the max stack for EVERY SINGLE stackable item?

Reply:
That's a very good question, which i don't have na answer 😢

Reply:
all default for stackable items are 30000
imho, I prefer we make a new table file (can be eac server later) to use max stack limit, example
```
-1 30000
<itemid> <stack_limit>
```
the stack info for now, RK's runes are 60 in inventory and 3 for AB's Ancilla in inventory
source: https://github.com/rathena/rathena/blob/master/db/re/item_stack.txt

Reply:
that's for inventory right? this for stackable storage, so no need a new table file right now ,i think

Reply:
@c4c1n6kr3m1 actually this piece of code is for any type of transference, 
* inventory to storage
* inventory to cart
* cart to inventory and so on

Reply:
my bad, didn't realize there is new processTransferItems